data/data-pipeline/data_pipeline/tests/test_score_calculation.py

```diff
@@ -1,302 +1,152 @@
-
-@pytest.fixture
-def low_income_threshold():
-    return ScoreNarwhal(final_score_df).LOW_INCOME_THRESHOLD
-
-
-@pytest.fixture
-def low_income_percentile():
-    return (
-        field_names.FPL_200_SERIES_IMPUTED_AND_ADJUSTED,
-        field_names.POVERTY_LESS_THAN_200_FPL_IMPUTED_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    )
-
-
-@pytest.fixture
-def low_hs_threshold():
-    return ScoreNarwhal(final_score_df).LACK_OF_HIGH_SCHOOL_MINIMUM_THRESHOLD
-
-
-@pytest.fixture
-def low_hs_percent():
-    return field_names.LOW_HS_EDUCATION_FIELD, field_names.HIGH_SCHOOL_ED_FIELD
-
-
-@pytest.fixture
-def env_percentile_threshold():
-    return ScoreNarwhal(final_score_df).ENVIRONMENTAL_BURDEN_THRESHOLD
-
-
-# Climate factor thresholds
-@pytest.fixture
-def expected_pop_loss_percentile_columns():
-    return [
-        field_names.EXPECTED_POPULATION_LOSS_EXCEEDS_PCTILE_THRESHOLD,
-        field_names.EXPECTED_POPULATION_LOSS_RATE_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    ]
-
-
-@pytest.fixture
-def expected_ag_loss_percentile_columns():
-    return [
-        field_names.EXPECTED_AGRICULTURAL_LOSS_EXCEEDS_PCTILE_THRESHOLD,
-        field_names.EXPECTED_AGRICULTURE_LOSS_RATE_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    ]
-
-
-@pytest.fixture
-def expected_bldg_loss_percentile_columns():
-    return [
-        field_names.EXPECTED_BUILDING_LOSS_EXCEEDS_PCTILE_THRESHOLD,
-        field_names.EXPECTED_BUILDING_LOSS_RATE_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    ]
-
-
-@pytest.fixture
-def expected_flood_percentile_columns():
-    return [
-        field_names.HIGH_FUTURE_FLOOD_RISK_FIELD,
-        field_names.FUTURE_FLOOD_RISK_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    ]
-
-
-@pytest.fixture
-def expected_wildfire_percentile_columns():
-    return [
-        field_names.HIGH_FUTURE_WILDFIRE_RISK_FIELD,
-        field_names.FUTURE_WILDFIRE_RISK_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-    ]
-
-
-@pytest.fixture
-def score_col():
-    return field_names.SCORE_N_COMMUNITIES
-
-
-@pytest.fixture
-def score_col_with_donuts():
-    return field_names.FINAL_SCORE_N_BOOLEAN
-
-
-@pytest.fixture
-def lead_paint_columns():
-    """Returns lead paint percentile column, Median house value percentile column,
-    and thresholded boolean column for the proxy (in order).
-    Does not return thresholds
-    """
-    return [
-        field_names.LEAD_PAINT_FIELD + field_names.PERCENTILE_FIELD_SUFFIX,
-        field_names.MEDIAN_HOUSE_VALUE_FIELD
-        + field_names.PERCENTILE_FIELD_SUFFIX,
-        field_names.LEAD_PAINT_PROXY_PCTILE_THRESHOLD,
-    ]
-
+@dataclass
+class PercentileTestConfig:
+    percentile_column_name: str
+    threshold_column_name: str
+    threshold: float
+    percentile_column_need_suffix: bool = True
 
-@pytest.fixture
-def lead_paint_proxy_home_value_threshold():
-    return ScoreNarwhal(final_score_df).MEDIAN_HOUSE_VALUE_THRESHOLD
+    @property
```

Comment:
This is muuuuch nicer and clearer.

Reply:
I don't love that I have that `percentile_column_need_suffix` flag but also like, it's good enough for smoketests?